test/test_gpu_utils.cpp

```diff
@@ -25,6 +25,44 @@
 
 namespace nvfuser {
 
+int myFavoriteFunction(int a, int b) {
+  DEBUG_PRINT_SCOPE(a, b);
+  if (a > 0) {
+    RECORD_AND_RETURN(a + b);
+  } else {
+    RECORD_AND_RETURN(a - b);
+  }
+}
+
+TEST_F(NVFuserTest, FunctionTrace) {
+  {
+    std::stringstream ss;
+    DebugStreamGuard g(ss);
+    DebugDumpOptionsGuard gg;
+    gg.getCurOptions().set(DebugDumpOption::FunctionTrace, {".*Favorite.*"});
+    EXPECT_EQ(myFavoriteFunction(1, 2), 3);
+    EXPECT_THAT(
+        ss.str(), ::testing::HasSubstr("Entering myFavoriteFunction(1, 2)"));
+    EXPECT_THAT(
+        ss.str(),
+        ::testing::HasSubstr("Leaving myFavoriteFunction returning 3 at "));
+    EXPECT_THAT(ss.str(), ::testing::HasSubstr("test_gpu_utils.cpp:31"));
+  }
+  {
+    std::stringstream ss;
+    DebugStreamGuard g(ss);
+    DebugDumpOptionsGuard gg;
+    gg.getCurOptions().set(DebugDumpOption::FunctionTrace, {".*Favorite.*"});
+    EXPECT_EQ(myFavoriteFunction(-1, 2), -3);
+    EXPECT_THAT(
+        ss.str(), ::testing::HasSubstr("Entering myFavoriteFunction(-1, 2)"));
+    EXPECT_THAT(
+        ss.str(),
+        ::testing::HasSubstr("Leaving myFavoriteFunction returning -3 at "));
+    EXPECT_THAT(ss.str(), ::testing::HasSubstr("test_gpu_utils.cpp:33"));
+  }
```

Comment:
Is it possible to separate this into another test? This way, tests can run in parallel and run in isolation.

Reply:
done